packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -566,8 +566,8 @@ export function flushRoot(root: FiberRoot, expirationTime: ExpirationTime) {
   flushSyncCallbackQueue();
 }
 
-export function flushInteractiveUpdates() {
-  if (workPhase === RenderPhase || workPhase === CommitPhase) {
+export function flushDiscreteUpdates() {
+  if (workPhase !== NotWorking) {
```

Comment:
Why did you change this? It should flush inside `unstable_batchedUpdates`, too.

Reply:
In the previous `interactiveUpdates` function, we only called `flushInteractiveUpdates` when `workPhase` was not `NotWorking`. So this was to duplicate that behaviour or some `act` tests fails. Would you prefer we had another version of this function that is only called on the event path and thus revert `flushDiscreteUpdates` back to the old conditional?

Reply:
Actually, I think I know what you meant now. Changes are reflected in latest commit.

Reply:
The purpose of the check is to handle nested event dispatches, like `element.click()` or `element.focus()`, that are invoked inside a React lifecycle. In that case `workPhase === CommitPhase`; we can't synchronously flush pending discrete updates in that case because we're already in the middle of another render. So we exit and do nothing. Arguably, we should defer the nested event dispatch to a microtask, but that would be a breaking change in legacy mode. That seems outside the scope of this PR, though.

However, there's no valid case for `workPhase` to be `RenderPhase`. Without the check, this will trigger an internal invariant, but we should have a user-facing error with a helpful message.

So this is what I would suggest:

```js
if (workPhase === CommitPhase) {
  // We're inside the commit phase, so we can't synchronously flush pending
  // work. This is probably a nested event dispatch triggered by a
  // lifecycle/effect, like `el.focus()`. Exit.
  return;
}

if (workPhase === RenderPhase) {
  invariant(
    false,
    'unstable_flushDiscreteUpdates: Cannot flush updates when React is '
    + 'already rendering.'
  );
}

// Proceed with flushing
```

Reply:
Oops didn't see your comment while I was typing :D